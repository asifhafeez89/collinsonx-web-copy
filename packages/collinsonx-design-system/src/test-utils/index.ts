import userEvent from '@testing-library/user-event';

export * from '@testing-library/react';
export { userEvent };
export { default as render, Provider } from './render';
