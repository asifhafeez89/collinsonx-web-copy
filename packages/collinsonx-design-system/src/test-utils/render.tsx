import { RenderOptions, render } from '@testing-library/react';
import { MantineProvider } from '@mantine/core';
import { ReactElement, ReactNode } from 'react';

export const Provider = ({ children }: { children: ReactNode }) => (
  <MantineProvider>{children}</MantineProvider>
);

type Render = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => ReturnType<typeof render>;

const customRender: Render = (ui, options) =>
  render(ui, {
    wrapper: Provider as unknown as RenderOptions['wrapper'],
    ...options,
  });

export default customRender;
