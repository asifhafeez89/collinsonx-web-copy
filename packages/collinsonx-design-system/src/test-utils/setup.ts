import userEvent from '@testing-library/user-event';
import render from './render';

export default function (jsx: Parameters<typeof render>[0]) {
  return {
    user: userEvent.setup(),
    ...render(jsx),
  };
}
