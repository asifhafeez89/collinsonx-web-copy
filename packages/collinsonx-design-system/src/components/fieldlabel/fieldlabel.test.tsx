import renderer from 'react-test-renderer';
import { Provider } from 'test-utils';
import FieldLabel from '.';

const mockFn = jest.fn();

describe('<FieldLabel />', () => {
  it('renders fieldlabel', () => {
    const tree = renderer
      .create(
        <Provider>
          <FieldLabel
            title="Time of arrival"
            value={'test'}
            handleClick={mockFn}
          />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
