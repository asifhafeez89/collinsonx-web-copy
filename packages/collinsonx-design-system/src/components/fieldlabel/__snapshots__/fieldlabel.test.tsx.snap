// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FieldLabel /> renders fieldlabel 1`] = `
[
  <style
    dangerouslySetInnerHTML={
      {
        "__html": "@media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}",
      }
    }
    data-mantine-styles="classes"
  />,
  .emotion-0 {
  border-radius: 10px;
  border: 1px solid #e9ecef;
  padding: 20px;
  margin-bottom: 10px;
}

<div
    className="emotion-0"
    onClick={[MockFunction]}
  >
    <h3>
      Time of arrival
    </h3>
    <p>
      test
    </p>
  </div>,
]
`;
